src/ffi/locale.rs

```diff
@@ -35,7 +35,11 @@ mod inner {
                 )
             };
 
-            let save = unsafe { libc::uselocale(utf8_locale) };
+            let save = if !utf8_locale.is_null() {
```

Comment:
I'd just add a small comment such as:

```rust
// `libc::newlocale(...)` may fail in case there is no locale information in the
// running system.
```